File: app/src/main/java/com/outplaysoftworks/sidedeck/AnimateTextView.java
package com.outplaysoftworks.sidedeck;

import android.animation.ValueAnimator;
import android.widget.TextView;

/**
 * Helper class to animate TextViews displaying integers in an odometer style
 */

class AnimateTextView {

    /**
     * Animates a TextView displaying integers odometer style
     * @param initialValue Value to start at
     * @param finalValue Value to finish animation at
     * @param textview Target TextView
     * @param isZero If true uses longer animation duration to accommodate longer sond effects
     */
    static void animateTextView(int initialValue, int finalValue, final TextView textview, boolean isZero) {
        if (initialValue != finalValue) { //will not do anything if both values are equal
            ValueAnimator valueAnimator = ValueAnimator.ofInt(initialValue, finalValue);
            if(isZero){
                valueAnimator.setDuration(AppConstants.LPCHANGEANIMATIONDURATIONLONG);
            } else{
                valueAnimator.setDuration(AppConstants.LPCHANGEANIMATIONDURATION);
            }
            valueAnimator.addUpdateListener(valueAnimator1 -> textview.setText(valueAnimator1.getAnimatedValue().toString()));
            valueAnimator.start();
        }
    }
}

File: app/src/main/java/com/outplaysoftworks/sidedeck/AnimateTextView.kt
package com.outplaysoftworks.sidedeck

import android.animation.ValueAnimator
import android.widget.TextView

/**
 * Helper class to animate TextViews displaying integers in an odometer style
 */

internal object AnimateTextView {

    /**
     * Animates a TextView displaying integers odometer style
     * @param initialValue Value to start at
     * @param finalValue Value to finish animation at
     * @param textview Target TextView
     * @param isZero If true uses longer animation duration to accommodate longer sound effects
     */
    fun animateTextView(initialValue: Int, finalValue: Int, textview: TextView, isZero: Boolean) {
        if (initialValue != finalValue) { //will not do anything if both values are equal
            val valueAnimator = ValueAnimator.ofInt(initialValue, finalValue)
            if (isZero) {
                valueAnimator.duration = AppConstants.LPCHANGEANIMATIONDURATIONLONG.toLong()
            } else {
                valueAnimator.duration = AppConstants.LPCHANGEANIMATIONDURATION.toLong()
            }
            valueAnimator.addUpdateListener { valueAnimator1 -> textview.text = valueAnimator1.animatedValue.toString() }
            valueAnimator.start()
        }
    }
}
